static/js/publisher/release/components/revisionsListRow.js

```diff
@@ -1,49 +1,74 @@
-import React, { Fragment } from "react";
+import React, { Fragment, useState } from "react";
 import PropTypes from "prop-types";
 import { connect } from "react-redux";
 
 import distanceInWords from "date-fns/distance_in_words_strict";
 import format from "date-fns/format";
 
-import { getChannelString } from "../../../libs/channels.js";
+import { getChannelString } from "../../../libs/channels";
 import { useDragging, DND_ITEM_REVISIONS, Handle } from "./dnd";
 import { toggleRevision } from "../actions/channelMap";
+import {
+  releaseRevision,
+  updateProgressiveReleasePercentage,
+  pauseProgressiveRelease,
+  resumeProgressiveRelease
+} from "../actions/pendingReleases";
 import {
   getSelectedRevisions,
   getProgressiveState,
   isProgressiveReleaseEnabled
 } from "../selectors";
 
 import RevisionLabel from "./revisionLabel";
-import { ProgressiveBar } from "./progressiveBar";
+import { InteractiveProgressiveBar } from "./progressiveBar";
 
 const RevisionsListRow = props => {
   const {
+    index,
     revision,
     isSelectable,
     showChannels,
     showBuildRequest,
     isPending,
     isActive,
     getProgressiveState,
-    isProgressiveReleaseEnabled
+    isProgressiveReleaseEnabled,
+    updateProgressiveReleasePercentage,
+    releaseRevision,
+    pauseProgressiveRelease,
+    resumeProgressiveRelease
   } = props;
 
+  const [canDrag, setDraggable] = useState(true);
+
   const revisionDate = revision.release
     ? new Date(revision.release.when)
     : new Date(revision.created_at);
 
   const isSelected = props.selectedRevisions.includes(revision.revision);
 
+  let channel;
+  if (revision.release) {
+    channel = getChannelString(revision.release);
+  }
+
   function revisionSelectChange() {
     props.toggleRevision(revision);
   }
 
   let progressiveState;
+  let previousRevision;
+  let pendingProgressiveState;
 
-  if (revision.release) {
-    progressiveState = getProgressiveState(
-      getChannelString(revision.release),
+  // Only show the progressive release status if it's the latest revision
+  if (index === 0 && revision.release) {
```

Comment:
This component doesn't care about `index` value. If the prop is meant to define if progressive release should be shown it should be `showProgressive` flag or something like this.